### PR TITLE
[extensions] Added the ability to add buttons to the toolbar

### PR DESCRIPTION
**Pull request by igorkorsukov**
[Screencast from 2024-09-20 11-42-12.webm](https://github.com/user-attachments/assets/4b565183-614e-42c6-b7f0-43f738da59e5)

### CODE REVIEW

**Comment by Jojo-Schmitz** [comment]
Do I see it right that this allows for a Save button?

**Comment by igorkorsukov** [comment]
> Do I see it right that this allows for a Save button to get added to the toolbar?

Yes ;)

**Comment by Jojo-Schmitz** [comment]
Nive!, That'll close #18883/#18896

File: src/framework/ui/view/iconcodes.h [hunk 1]
```diff
@@ -447,6 +448,15 @@ class IconCode
     };
 
     Q_ENUM(Code)
+
+    static Code fromString(const char* key)
+    {
+        int enumIndex = staticMetaObject.indexOfEnumerator("Code");
+        QMetaEnum me = staticMetaObject.enumerator(enumIndex);
```
[the remark this answers is not in the document]
Comment (igorkorsukov) [reply src/framework/ui/view/iconcodes.h]: fixed

File: src/framework/extensions/qml/Muse/Extensions/ExtensionsToolBar.qml
[the remark this answers is not in the document]
Comment (igorkorsukov) [reply src/framework/extensions/qml/Muse/Extensions/ExtensionsToolBar.qml:1]: implementation changed to using StyledToolBarView

**Review by cbjeukendrup — APPROVED** [review]

**Pull request merged**